### PR TITLE
CSF: Refactor router utils into CSF library

### PR DESCRIPTION
**Pull request by shilman**
Issue: N/A

Prep work for #9098 

## What I did

- [x] Move CSF-related functions into `@storybook/csf`
- [x] Re-export functions from router/utils after CI passes

## How to test

See CI results

### CODE REVIEW

**Review by tmeasday — APPROVED** [review]
LGTM

**Review by ndelangen — APPROVED** [review]

**Comment by Hypnosphi** [comment]
@shilman Why you had to remove it from monorepo?

**Comment by shilman** [comment]
@Hypnosphi I want to version it independently of Storybook, since third-party adopters will contribute as part of a standardization process.

**Comment by Hypnosphi** [comment]
Can it stay in the repo, but not in the lerna publish config?

File: lib/router/src/utils.ts [hunk 1]
```diff
@@ -1,43 +1,16 @@
 import qs from 'qs';
 import memoize from 'memoizerific';
-import startCase from 'lodash/startCase';
+
+// FIXME: Remove in SB 6.0
+export { toId, parseKind, sanitize } from '@storybook/csf';
```
Comment (Hypnosphi) [inline lib/router/src/utils.ts]: @shilman Why did you have to remove `storyNameFromExport` and other exports that are present in [`5.2`](https://github.com/storybookjs/storybook/blob/master/lib/router/src/utils.ts)?

Comment (shilman) [reply lib/router/src/utils.ts]: @Hypnosphi That was accidental. I'll add `storyNameFromExport` back. That's the only one, right?

Comment (Hypnosphi) [reply lib/router/src/utils.ts]: That's the only one I used =) And probably the only one at all

**Pull request merged**